### PR TITLE
Add pre-commit hook for format checks

### PR DESCRIPTION
**Pull request by aledomu**
It lacks lint checks. I need to know what commands you are running for this.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Sorry but I'm strongly against this. :-/ We're about to switch to github actions and there it'll be very quick to get errors for formatting. I don't want anything to be done on the user's side if possible.

**Comment by aledomu** [comment]
Just curious, when forking the repo as a contributor, will actions be imported? Or will actions only work when doing the PR as it does right now?

File: hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
```
Comment (sdroege) [inline hooks/pre-commit:1]: This won't work on Windows

File: hooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+if ! which rustup &> /dev/null; then
+    curl https://sh.rustup.rs -sSf  | sh -s -- -y
```
Comment (sdroege) [inline hooks/pre-commit:4]: Not everybody uses rustup :)

File: hooks/pre-commit (hunk 1 above)
Comment (aledomu) [reply hooks/pre-commit:1]: :facepalm:

**Comment by aledomu** [comment]
It seems clear this isn't a good idea at all. Closing

**Comment by sdroege** [comment]
> It seems clear this isn't a good idea at all. Closing

It's a good idea to have something like this but there's a reason why I asked how you want to do that in the issue :) All ways of doing this that I could think of are going to break in various situations.

**Pull request closed without merging**